src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonesArr = [];
+  let stateCopy = { ...state };
+  let addCopy;
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        const addProps = obj.extraData;
+
+        addCopy = {
+          ...stateCopy,
+          ...addProps,
+        };
+
+        clonesArr.push(addCopy);
+        stateCopy = Object.assign({}, addCopy);
+
+        break;
+
+      case 'removeProperties':
+        const removeProps = obj.keysToRemove;
+
+        for (const key of Object.keys(stateCopy)) {
+          if (removeProps.indexOf(key) !== -1) {
+            delete stateCopy[key];
+          }
+        }
+
+        clonesArr.push(stateCopy);
+        break;
+
+      default:
+        const emptyObj = {};
+
+        stateCopy = Object.assign({}, emptyObj);
```

Comment:
why do you need to assign an empty object to another empty object?

Reply:
better delete all properties from the stateCopy to keep the empty state for the next iterations